pkg/tecdsa/signing/protocol_test.go

```diff
@@ -234,22 +256,210 @@ func TestGenerateSymmetricKeys_InvalidEphemeralPublicKeyMessage(t *testing.T) {
+	members, err := initializeTssRoundOneMembersGroup(
+		dishonestThreshold,
+		groupSize,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Perform TSS round one for each group member.
+	for _, member := range members {
+		// To simulate the outgoing message timeout we do two things:
+		// - we pass an already cancelled context
+		// - we make sure no message is emitted from the channel by overwriting
+		//   the existing channel with a new one that won't receive any
+		//   messages from the underlying TSS local party
+		ctx, cancelCtx := context.WithCancel(context.Background())
+		cancelCtx()
+		member.tssOutgoingMessagesChan = make(<-chan tss.Message)
+
+		_, err := member.tssRoundOne(ctx)
+
+		expectedErr := fmt.Errorf(
+			"TSS round one outgoing messages were not generated on time",
+		)
+		if !reflect.DeepEqual(expectedErr, err) {
+			t.Errorf(
+				"unexpected error for member [%v]\n"+
+					"expected: %v\n"+
+					"actual:   %v\n",
+				member.id,
+				expectedErr,
+				err,
+			)
+		}
+	}
+}
+
+func TestTssRoundOne_SymmetricKeyMissing(t *testing.T) {
+	members, err := initializeTssRoundOneMembersGroup(
+		dishonestThreshold,
+		groupSize,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Perform TSS round one for each group member.
+	for _, member := range members {
+		// Cleanup symmetric key cache.
+		member.symmetricKeys = make(map[group.MemberIndex]ephemeral.SymmetricKey)
+
+		ctx, cancelCtx := context.WithTimeout(context.Background(), 10*time.Second)
+
+		_, err := member.tssRoundOne(ctx)
+
+		if !strings.Contains(
+			err.Error(),
+			"cannot get symmetric key with member",
+		) {
+			t.Errorf("wrong error for member [%v]: [%v]", member.id, err)
+		}
+
+		cancelCtx()
+	}
+}
+
+func assertOutgoingAggregateMessage(
+	t *testing.T,
+	message struct{
+		senderID group.MemberIndex
+		broadcastPayload []byte
+	    peersPayload map[group.MemberIndex][]byte
+		sessionID string
+	},
+	memberID group.MemberIndex,
+	memberSymmetricKeys map[group.MemberIndex]ephemeral.SymmetricKey,
+) {
+	// We should always be the sender of our own messages.
+	testutils.AssertIntsEqual(
+		t,
+		fmt.Sprintf(
+			"message sender in message generated by member [%v]",
+			memberID,
+		),
+		int(memberID),
+		int(message.senderID),
+	)
+
+	// We should always generate a broadcast payload.
+	if len(message.broadcastPayload) == 0 {
+		t.Errorf(
+			"empty broadcast payload in message generated by member [%v]",
+			memberID,
+		)
+	}
+
+	// We should always generate groupSize-1 of peers payloads.
+	testutils.AssertIntsEqual(
+		t,
+		fmt.Sprintf(
+			"count of peers payloads in message "+
+				"generated by member [%v]",
+			memberID,
+		),
+		groupSize-1,
+		len(message.peersPayload),
+	)
+
+	// Each P2P payload should be encrypted using the proper symmetric key.
+	for receiverID, encryptedPayload := range message.peersPayload {
+		symmetricKey := memberSymmetricKeys[receiverID]
+		if _, err := symmetricKey.Decrypt(encryptedPayload); err != nil {
```

Comment:
Are we sure it's producing an error and not some mojibake?

Reply:
This is what happens under the hood: https://github.com/keep-network/keep-common/blob/b87c2a9c66090c172dbd4ab4444ef80f206e603e/pkg/encryption/box.go#L48. It returns an error in case of a broken ciphertext. See also https://github.com/keep-network/keep-core/blob/84b8fd8d3e582bd6f0985e113c3bea67bf2935bc/pkg/crypto/ephemeral/symmetric_key_test.go#L69